### PR TITLE
[Exec] storehouse - a fork-aware flexible payload store - part 2

### PR DESCRIPTION
**Pull request by ramtinms**
Storehouse is a wrapper over any persistent payload storage
- adding in-memory functionality to keep the results of unfinalized blocks in memory (fork-awareness)
- providing a fast reader interface for executing transactions 
- consuming block finalization signals and moving data from in-memory results to persistent storage.  

This is the second part of cargo PRs which defines the persistent storage interface and provides an in-memory implementation which could be used for testing.

### CODE REVIEW

File: engine/execution/state/cargo/storage/storage.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+// Storage is a block height-based persistant storage for storing historic register values.
+// it accepts updates as a set of sequential fork-free commit calls and
+// let user to query for the value of a register for at a given height
+type Storage interface {
+	// RegisterValueAt returns the register value at the given height
+	// note that if a register is not updated at specific block it pretaining the previous value
+	// non-existing registers return nil value by default and errors are only kept for fatal ones
+	RegisterValueAt(height uint64, blockID flow.Identifier, id flow.RegisterID) (value flow.RegisterValue, err error)
```
Comment (pattyshack) [inline engine/execution/state/cargo/storage/storage.go:14]: I think i've commented on this before, but it seems better to have the storage return a "view" interface, which in turn exposes the register access method.  e.g.,
```
  ViewAt(height uint64, blockID Identifier) View
  ...

type View interface {
  Register(id RegisterID)
}
```

the indirection simplifies the access method signature, and the view implementation can cache/reuse the height / block lookup

Comment (ramtinms) [reply engine/execution/state/cargo/storage/storage.go:14]: I used it for the external facing one [here](https://github.com/onflow/flow-go/pull/4248/files#diff-7460c6b22289104d07aa04b5db0cedafba9033ae237e1073ad2ef4b2684e960eR42) but I can have it to the lower level one here as well.

Comment (ramtinms) [reply engine/execution/state/cargo/storage/storage.go:14]: I think what might have been misleading is the name storage, I should probably use that one for the actual storage and use this internal one with a different name

**Comment by pattyshack** [comment]
sounds like leo is taking over.  i'll remove myself for now.

**Comment by ramtinms** [comment]
Obsolete branch, going to close this PR.

**Pull request closed without merging**